### ISSUE TITLE
Improvements of hnsnetwork

### ISSUE THREAD

**Issue by randomlogin** [opensystm/handshake-micro-grants#10]
Hello,
I'm Alex, one of the creators of hnsnetwork block explorer and (very) sporadic contributor to the handshake codebase.

What we have built previously?
[hnsnetwork.com](https://hnsnetwork.com) an independent (not using internal hsd DB, thus surviving when something happens to the most of the
nodes) indexer/block explorer which is open source and has been running since 2021.

What will be implemented? What the grant will go for?
There is a long list of potential improvements which cannot be done in just month and would require more funding, so we
came up with a realistic list:

- working mobile version
- new pages 
   - network stats as the whole (nodes, versions)
   - other stats lists: toplists of the most transferred domains, the most frequent record changed, the most rich addresses, etc.
   - shortly ending auctions, domains soon to expire
- fix/optimisation of mempool (right now is too big for a single request)
- rework of tx page, anchoring to exact outputs and txs when navigating
- search improvements (if there is one result, then it redirects to it; also bulk search)
- optimization of big requests (e.g. addresses with tens of thousands of outputs)
- some backend/server optimization/bugfixes to prevent rare downtime moments
- server cost cover
- small improvements (emoji urls, not found page, cards, spinner, several auctions for the same name, etc.)
- export to csv

Completion details.
The milestones will be done by the team from @handshake-labs (namely @randomlogin and @v-kabashnikov). 
We estimate that a month should suffice for all the points from the above.

Why it is important?
Having an independent source of information is crucial (especially technical information), also I think it's the only working open source explorer.

**Comment by handshake-enthusiast** [comment]
Hi Alex! Great to have more block explorers with more features.

>also I think it's the only working open source explorer

Just wanted to share a few more examples, so you can get some inspiration from them for your block explorer:  

| URL                                   | Source Code                                         |
|---------------------------------------|-----------------------------------------------------|
| [https://e.hnsfans.com/](https://e.hnsfans.com/) | [dxpool/hns-explorer](https://github.com/dxpool/hns-explorer) |
| [https://blockexplorer.com/](https://blockexplorer.com/) | [kurumiimari/hsexplorer](https://github.com/kurumiimari/hsexplorer) |
| [https://hnscan.com/](https://hnscan.com/)       | [HNScan/HNScan](https://github.com/HNScan/HNScan)               |
| [https://hnsxplorer.com/](https://hnsxplorer.com/)   | [tdickman/handshake-explorer](https://github.com/tdickman/handshake-explorer) |
| [https://3xpl.com/handshake](https://3xpl.com/handshake) | [3xplcom/Core](https://github.com/3xplcom/Core)                   |

**Comment by randomlogin** [comment]
@handshake-enthusiast , thanks for the list, I didn't know some of them are open sourced.
Though I think blockexplorer.com, hnscan.com and hnsxplorer.com do not work, at least for me.

**Comment by handshake-enthusiast** [comment]
@randomlogin indeed, those three block explorers seem to be not supported anymore. However, due to their open source nature someone else can get them up and running! If you want to take a look at them in the past you can use https://web.archive.org. E.g. see the links below:  
- https://web.archive.org/web/20210815172716/https://blockexplorer.com/
- https://web.archive.org/web/20210219062449/https://hnscan.com/
- https://web.archive.org/web/20200528013115/https://hnsxplorer.com/
